packages/graphql/src/services/sessions.ts

```diff
@@ -362,17 +360,82 @@ export async function deactivateSessionBlock(
   )
 
   const redisMulti = ctx.redisExec.multi()
+  redisMulti.hgetall(`s:${sessionId}:lb`)
+  redisMulti.hgetall(`s:${sessionId}:b:${sessionBlockId}:lb`)
   activeInstanceIds.forEach((instanceId) => {
-    redisMulti.hgetall(`s${sessionId}:i:${instanceId}:results`)
+    redisMulti.hgetall(`s:${sessionId}:i:${instanceId}:responseHashes`)
+    redisMulti.hgetall(`s:${sessionId}:i:${instanceId}:responses`)
+    redisMulti.hgetall(`s:${sessionId}:i:${instanceId}:results`)
   })
-  const allInstanceResults = await redisMulti.exec()
+  const cachedResults = await redisMulti.exec()
 
-  console.warn(allInstanceResults)
+  if (!cachedResults) return session
 
-  return session
+  const mappedResults: any[] = cachedResults.map(([_, result]) => result)
+
+  const sessionLeaderboard: Record<string, string> = mappedResults[0]
+  const blockLeaderboard: Record<string, string> = mappedResults[1]
+
+  const instanceResults: Record<
+    string,
+    {
+      responseHashes: Record<string, string>
+      responses: Record<string, string>
+      results: Record<string, any>
+    }
+  > = mappedResults.slice(2).reduce((acc: any, cacheObj: any, ix) => {
+    const ixMod = ix % 3
+    const instance: QuestionInstance =
+      session.activeBlock!.instances[Math.floor((ix - ixMod) / 3)]
+    switch (ixMod) {
+      // results
+      case 2: {
```

Comment:
would be much nicer to have types defined for the cacheObj depending on the case here, but not implemented right now